src/server/common/resultsets/game_history.cpp

```diff
@@ -52,6 +52,13 @@ void game_history::read(mariadb::result_set_ref rslt)
 std::unique_ptr<simple_wml::document> game_history::to_doc()
 {
     auto doc = std::make_unique<simple_wml::document>();
+
+    if(results.size() == 0)
+    {
+        doc->set_attr_dup("no_data", "yes");
```

Comment:
Why is this necessary? Just no `[game_history_result]` means the same thing, no? If more structure is needed then the existing WML should be wrapped in `[game_history_results]` or so. (Certainly the way just an error attribute is sent right now is not so great.)

Reply:
If an unrelated response is sent from the server to the client, then it would also have no [game_history_result].

Reply:
If an unrelated response contains a `no_data` attribute then you have a similar issue. Thus the suggestion to always send a tag like `[game_history_results]`.

Reply:
I will update this then once the PR for storing a leader's information in the database has been resolved.